ssg/controls.py

```diff
@@ -202,9 +202,16 @@ def get_all_controls_of_level(self, policy_id, level_id):
 
         all_policy_controls = self.get_all_controls(policy_id)
         eligible_controls = []
-        for c in all_policy_controls:
-            if len(level_ids.intersection(c.levels)) > 0:
-                eligible_controls.append(c)
+        defined_variables = []
+        # we will go level by level, from top to bottom
+        # this is done to enable overriding of variables by higher levels
+        for lv in level_ids:
```

Comment:
`level_ids` is a `set`, so we cannot assume any particular order over the IDs.

Additionally, we cannot assume that the levels returned by `get_level_with_ancestors` is ordered, it also returns the levels in a `set`.